### PR TITLE
Add the protocol option in exporting pickle.

### PR DESCRIPTION
**Pull request by grigorisg9gr**
This is a useful option, allowing to export large files. 
A more thorough explanation of the different pickle protocols in https://docs.python.org/3/library/pickle.html .

### CODE REVIEW

**Comment by patricksnape** [comment]
This seems good to me, @jabooth happy?

File: menpo/io/output/base.py [hunk 1]
```diff
@@ -113,6 +113,8 @@ def export_pickle(obj, fp, overwrite=False):
         The string path or file-like object to save the object at/into.
     overwrite : `bool`, optional
         Whether or not to overwrite a file if it already exists.
+    protocol : `int`, optional
+        Save with a different pickle protocol the file. Mostly for python 3.
```
Comment (nontas) [inline menpo/io/output/base.py]: It would be good to add some kind of table here in the doc that explains what each protocol int value means.

Comment (patricksnape) [inline menpo/io/output/base.py]: @nontas Sounds sensible.

**Comment by patricksnape** [comment]
Great, thanks @grigorisg9gr

**Comment by nontas** [comment]
+1

File: menpo/io/output/base.py [hunk 2]
```diff
@@ -113,7 +113,22 @@ def export_pickle(obj, fp, overwrite=False):
         The string path or file-like object to save the object at/into.
     overwrite : `bool`, optional
         Whether or not to overwrite a file if it already exists.
-
+    protocol : `int`, optional
+        Save with a different pickle protocol the file.
```
Comment (jabooth) [inline menpo/io/output/base.py]: Change this line to just 'The Pickle protocol used to serialize the file'. The 'change default' but is redundant - that's implied by the fact it's an optional kwarg. Picky I know! 🙊

**Comment by jabooth** [comment]
Sorry for the delayed review guys!

Feels like a bandaid around a larger problem (at some point for large models HDF5 might does start making sense) but can't argue with it being a pragmatic addition.

If we ever had the guts to go all in on Python 3.4+ one day you could revisit this and crack it up to 4 by default too, just a thought.

Anyway just that one doc line fix then we'll bring it in. Thanks @grigorisg9gr!

**Pull request merged**